redocly/scripts/combine-yamls.py

```diff
@@ -0,0 +1,136 @@
+                value = replace_br_tags(value)
+                value = escape_ampersands(value)
+                content[key] = value
+            else:
+                process_descriptions(value)
+    elif isinstance(content, list):
+        for item in content:
+            process_descriptions(item)
+
+def collect_openapi_components(file_paths, excluded_tags):
+    collected_data = {
+        "paths": {},
+        "schemas": {},
+        "securitySchemes": {},
+        "tags": {},
+        "x-tagGroups": {}
+    }
+
+    for group_name, file_path in file_paths.items():
+        if not os.path.exists(file_path):
+            print(f"Skipping missing file: {file_path}")
+            continue
+
+        with open(file_path, 'r') as content_file:
+            parsed_content = yaml.safe_load(content_file.read())
+
+        process_descriptions(parsed_content)
+
+        paths = parsed_content.get('paths', {})
+        collected_data["paths"].update(paths)
+
+        components = parsed_content.get('components', {})
+        schemas = components.get('schemas', {})
+        collected_data["schemas"].update(schemas)
+
+        security_schemes = components.get('securitySchemes', {})
+        collected_data["securitySchemes"].update(security_schemes)
+
+        tags = parsed_content.get('tags', [])
+        for tag in tags:
+            if group_name in excluded_tags and tag['name'] in excluded_tags[group_name]:
+                continue
+
+            original_tag_name = tag['name']
+            display_name = transform_tag_name(original_tag_name)
+            tag['x-displayName'] = display_name
+            collected_data["tags"][original_tag_name] = tag
+
+            group_name_transformed = transform_tag_group_name(group_name)
+            if group_name_transformed not in collected_data["x-tagGroups"]:
+                collected_data["x-tagGroups"][group_name_transformed] = {"name": group_name_transformed, "tags": []}
+
+            if original_tag_name not in collected_data["x-tagGroups"][group_name_transformed]["tags"]:
+                collected_data["x-tagGroups"][group_name_transformed]["tags"].append(original_tag_name)
+
+    if "DataUpload" not in collected_data["tags"]:
+        collected_data["tags"]["DataUpload"] = {
+            "name": "DataUpload",
+            "description": "Operations related to data upload.",
+            "x-displayName": "Data Upload"
+        }
+
+        data_in_group_name_transformed = transform_tag_group_name('data_in')
+        if data_in_group_name_transformed not in collected_data["x-tagGroups"]:
+            collected_data["x-tagGroups"][data_in_group_name_transformed] = {"name": data_in_group_name_transformed, "tags": []}
+        if "DataUpload" not in collected_data["x-tagGroups"][data_in_group_name_transformed]["tags"]:
+            collected_data["x-tagGroups"][data_in_group_name_transformed]["tags"].append("DataUpload")
+
+    return collected_data
+
+def merge_openapi_components(collected_data):
```

Comment:
The individual yaml files can't contain conflicting component names. No such assurances exist when merging multiple files. You may need to keep track of a symbol table to error out if the same component name is used in >1 file